src/main/java/com/rackspace/ceres/app/services/DataWriteService.java

```diff
@@ -138,4 +147,21 @@ private void cleanTags(Map<String, String> tags) {
         .retryWhen(appProperties.getRetryInsertDownsampled().build())
         .checkpoint();
   }
+
+  private Mono<?> storeMetricGroup(String tenant, Metric metric) {
+    String updatedAt = metric.getTimestamp().toString();
+    String metricGroup = metric.getTags().get(LABEL_METRIC_GROUP);
+    String metricName = metric.getMetric();
+    return metadataService.metricGroupExists(tenant, metricGroup).flatMap(exists -> {
+      if (exists) {
+        // Cassandra does not support NOT CONTAINS so we delete the metric name
```

Comment:
I think using `set` data type in cassandra would be more useful for this scenario than `list` type. As Set would not allow duplicates and single update query would work

Reply:
For reference:- https://docs.datastax.com/en/cql-oss/3.3/cql/cql_using/useInsertSet.html